File: 3/ts/dogs.ts
/**
 * This is the standard index.ts starting point.
 *
 * author Dominic M.
 * version 1.0
 * since 2024-10-16
 */
import { MrCoxallStack } from './MrCoxallStack.ts'

// Use MrCoxallStack class
const doggyPile = new MrCoxallStack()
doggyPile.pushString('red')
doggyPile.pushString('green')
doggyPile.pushString('blue')
console.log(`Dog items: ${doggyPile.getStack}`)
console.log(`The size of the stack is: ${doggyPile.getSize}`)
console.log(`Is the stack empty? ${doggyPile.getEmpty.toString()}`)
while (doggyPile.getSize > 0) {
  const topDog = doggyPile.popItem()
  console.log(`Popped dog item: ${topDog}`)
}
console.log(`Dog items: ${doggyPile.getStack}`)
console.log(`The size of the stack is: ${doggyPile.getSize}`)
console.log(`Is the stack empty? ${doggyPile.getEmpty.toString()}`)
console.log()

File: 3/java/Dogs.java
/*
 * The Starting point for the script.
 *
 * @author Dominic M.
 * @since 2024-10-22
 * @version 1.0
 */

/**
 * This is the Dogs class.
 */
final class Dogs {

    /**
     * Prevent instantiation.
     *
     * @throws IllegalStateException if this is ever called
     *
     */
    private Dogs() {
        throw new IllegalStateException("Cannot be instantiated");
    }

    /**
     * This is the main function.
     *
     * @param args no args used
     */
    public static void main(final String[] args) {

        // Use MrCoxallStack class
        final MrCoxallStack doggyPile = new MrCoxallStack();
        doggyPile.pushString("poodle");
        doggyPile.pushString("retriever");
        doggyPile.pushString("beagle");
        System.out.println("Dog items: " + doggyPile.getStack());
        System.out.println("The stack's size is: " + doggyPile.getSize());
        System.out.println(
            "Is the stack empty? " + String.valueOf(doggyPile.getEmpty())
        );
        while (doggyPile.getSize() > 0) {
            final String topDog = doggyPile.popItem();
            System.out.println("Popped dog item: " + topDog);
        }
        System.out.println("Dog items: " + doggyPile.getStack());
        System.out.println("The stack's size is: " + doggyPile.getSize());
        System.out.println(
            "Is the stack empty? " + String.valueOf(doggyPile.getEmpty())
        );

        System.out.println("\nDone.");
    }
}
